pennylane/io/qualtran_io.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the adapter class for Qualtran-PennyLane interoperability.
+"""
+# pylint:disable=
+from qualtran import (
+    Bloq,
+    CompositeBloq,
+    Soquet,
+    LeftDangle,
+    Side,
+    DecomposeNotImplementedError,
+    DecomposeTypeError,
+)
+
+import numpy as np
+import pennylane as qml
+
+from pennylane.operation import Operation
+from pennylane.wires import WiresLike
+
+
+def get_bloq_registers_info(bloq):
+    """Returns a `qml.registers` object associated with all named and unnamed registers and wires
+    in the bloq.
+
+    Args:
+        bloq: the bloq to get the registers info of
+
+    Returns:
+        dict: A dictionary that has all the named and un-named registers with default wire
+        ordering.
+
+    **Example**
+
+    Given a qualtran bloq:
+
+    from qualtran.bloqs.basic_gates import Swap
+
+    >>> qml.get_bloq_registers_info(Swap(3))
+    {'x': Wires([0, 1, 2]), 'y': Wires([3, 4, 5])}
+    """
+
+    cbloq = bloq.decompose_bloq() if not isinstance(bloq, CompositeBloq) else bloq
+
+    temp_register_dict = {}
+    for reg in cbloq.signature.rights():
+        temp_register_dict[reg.name] = reg.bitsize
```

Comment:
could be a comprehension and use the below`_get_named_registers` function to reduce duplicated code.